CONTRIBUTING.md

```diff
@@ -0,0 +1,139 @@
+# Local Development
+
+To develop & test repo2docker locally, you need:
+
+1. Familiarity with using a command line terminal
+2. A computer running macOS / Linux
+3. Some knowledge of git
+4. At least python 3.4
+5. Your favorite text editor
+6. A recent version of [Docker Community Edition](https://www.docker.com/community-edition)
+
+## Clone the repository
+
+First, you need to get a copy of the repo2docker git repository on your local
+disk.
+
+```bash
+git clone https://github.com/jupyter/repo2docker
+```
+
+This will clone repo2docker into a directory called `repo2docker`. You can
+make that your current directory with `cd repo2docker`.
+
+## Set up local virtual environment
+
+After cloning the repository (or your fork of the repo), you should set up an
+isolated environment to install libraries required for running / developing 
+repo2docker. There are many ways to do this, and a `virtual environment` is
+one of them.
+
+```bash
+python3 -m venv .
+source bin/activate
+pip3 install -e .
+pip3 install -r dev-requirements.txt
+```
+
+This should install all the libraries required for testing & running repo2docker!
+
+## Verify that docker is installed and running
+
+If you do not already have [Docker](https://www.docker.com/), you should be able
+to download and install it for your operating system using the links from the
+[official website](https://www.docker.com/community-edition). After you have
+installed it, you can verify that it is working by running the following commands:
+
+```bash
+docker version
+```
+
+It should output something like:
+
+```
+Client:
+ Version:      17.09.0-ce
+ API version:  1.32
+ Go version:   go1.8.3
+ Git commit:   afdb6d4
+ Built:        Tue Sep 26 22:42:45 2017
+ OS/Arch:      linux/amd64
+
+Server:
+ Version:      17.09.0-ce
+ API version:  1.32 (minimum version 1.12)
+ Go version:   go1.8.3
+ Git commit:   afdb6d4
+ Built:        Tue Sep 26 22:41:24 2017
+ OS/Arch:      linux/amd64
+ Experimental: false
+```
+
+Then you are good to go!
+
+## Running tests
+
+We have a lot of tests for various cases supported by repo2docker in the `tests/`
+subdirectory. These use [py.test](https://docs.pytest.org/).
+
+You can run all the tests with:
+
+```bash
+py.test -s tests/*
```

Comment:
You can run `pytest` with no args from the top level directory. Simpler because less args people might worry about what they mean?

Reply:
That doesn't work for me, it tries to run tests for all libraries installed in myvirtual environment, with hundreds of errors like:

```
ImportError while importing test module '/home/yuvipanda/code/repo2docker/lib64/python3.6/site-packages/ipython_genutils/tests/test_text.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
lib/python3.6/site-packages/ipython_genutils/tests/test_text.py:13: in <module>
    import nose.tools as nt
E   ModuleNotFoundError: No module named 'nose'
```

Reply:
Interesting. Magic. Let's keep it like this then.

Reply:
That's odd, if you're at the repo2docker package level it shouldn't run it on all of your packages. Could you mean something different from "top level directory"?